File: ai-interview/frontend/src/types/resume.ts
export interface Resume {
  id: string;
  title: string;
  original_filename: string;
  file_size: number;
  content_type?: string | null;
  parse_status: string;
  parse_error?: string | null;
  analysis_status: 'none' | 'success' | 'failed' | 'outdated';
  latest_overall_score?: number | null;
  high_risks: string[];
  is_default: boolean;
  created_at: string;
  updated_at: string;
}

export interface ResumeDetail extends Resume {
  parsed_text: string;
}

export interface ResumeSuggestion {
  priority: string;
  problem: string;
  advice: string;
  example?: string | null;
}

export interface ResumeSectionReview {
  section: string;
  score: number;
  comment: string;
}

export interface ResumeDiagnostic {
  id: string;
  resume_id: string;
  provider: string;
  model?: string | null;
  fallback_used: boolean;
  fallback_reason?: string | null;
  overall_score: number;
  summary: string;
  strengths_json: string[];
  weaknesses_json: string[];
  suggestions_json: ResumeSuggestion[];
  section_reviews_json: ResumeSectionReview[];
  follow_up_questions_json: string[];
  likely_interview_questions?: string[];
  raw_result_json: Record<string, unknown>;
  created_at: string;
}

File: ai-interview/backend/app/schemas/resume.py
from datetime import datetime

from pydantic import BaseModel, Field


class ResumeUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=255)
    parsed_text: str | None = Field(default=None, min_length=10)
    is_default: bool | None = None


class ResumeOut(BaseModel):
    id: str
    title: str
    original_filename: str
    file_size: int
    content_type: str | None = None
    parse_status: str
    parse_error: str | None = None
    analysis_status: str = "none"
    latest_overall_score: int | None = None
    high_risks: list[str] = Field(default_factory=list)
    is_default: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ResumeDetail(ResumeOut):
    parsed_text: str


class ResumeDiagnosticOut(BaseModel):
    id: str
    resume_id: str
    provider: str
    model: str | None = None
    fallback_used: bool = False
    fallback_reason: str | None = None
    overall_score: int
    summary: str
    strengths_json: list
    weaknesses_json: list
    suggestions_json: list
    section_reviews_json: list
    follow_up_questions_json: list
    raw_result_json: dict
    created_at: datetime

    model_config = {"from_attributes": True}


class ResumeDiagnoseRequest(BaseModel):
    llm_config_id: str | None = None
